tests/msbuild/Xamarin.MacDev.Tasks.Tests/TaskTests/MTouchTaskTests.cs

```diff
@@ -9,7 +10,7 @@
 
 namespace Xamarin.iOS.Tasks
 {
-	class CustomMTouchTask : MTouchTaskBase
+	class CustomMTouchTask : MTouch
```

Comment:
Why this change?

Reply:
The `GetAdditionalItemsToBeCopied` I modified (and added tests to) is not part of `MTouchTaskBase`, only from the`MTouch` subclass.

@emaf will correct me if needed :)

`*Base` tasks generally have all the logic required for running on the Mac.

The version running on Windows subclass them to add the behaviour required to remote the execution.

The version running on Mac also subclass `*Base` but it's empty (on purpose). That makes the msbuild target files identical (same names for tasks).

Reply:
That's correct, but there are no more empty subclasses now :). Because of .NET6 we "unified" both task implementations (Windows and Mac) on this repo, but what Sebastien said is still valid:
- Base classes have the core logic of the task
- The unified subclass contains code that enables running the task remotely (from Windows), but if it's being executed from a Mac it directly executes the base class https://github.com/xamarin/xamarin-macios/blob/main/msbuild/Xamarin.iOS.Tasks/Tasks/MTouch.cs#L16-L17. MTouch is one of the most complex tasks we have, ACTool is another good example because it's way simpler https://github.com/xamarin/xamarin-macios/blob/main/msbuild/Xamarin.iOS.Tasks/Tasks/ACTool.cs#L10-L13. ACTool does not implement ITaskCallback so there's no implementation for `GetAdditionalItemsToBeCopied` in that case, that's just for customizing what files need to be copied when executing remotely.

Reply:
Ah thank you so much @spouliot and @emaf for those very detailed responses! Makes more sense now!